keen/api.py

```diff
@@ -197,6 +197,231 @@ def get_all_collections(self):
 
         return response.json()
 
+    @requires_key(KeenKeys.MASTER)
+    def create_access_key(self, name, is_active=True, permitted=[], options={}):
+        """
+        Creates a new access key. A master key must be set first.
+
+        :param name: the name of the access key to create
+        :param is_active: Boolean value dictating whether this key is currently active (default True)
+        :param permitted: list of strings describing which operation types this key will permit
+                          Legal values include "writes", "queries", "saved_queries", "cached_queries",
+                          "datasets", and "schema".
+        :param options: dictionary containing more details about the key's permitted and restricted
+                        functionality
+        """
+
+        url = "{0}/{1}/projects/{2}/keys".format(self.base_url, self.api_version, self.project_id)
+        headers = utilities.headers(self.master_key)
+
+        payload_dict = {
+            "name": name,
+            "is_active": is_active,
+            "permitted": permitted,
+            "options": options
+        }
+        payload = json.dumps(payload_dict)
+
+        response = self.fulfill(HTTPMethods.POST, url, data=payload, headers=headers, timeout=self.get_timeout)
+        self._error_handling(response)
+        return response.json()
+
+    @requires_key(KeenKeys.MASTER)
+    def list_access_keys(self):
+        """
+        Returns a list of all access keys in this project. A master key must be set first.
+        """
+        url = "{0}/{1}/projects/{2}/keys".format(self.base_url, self.api_version, self.project_id)
+        headers = utilities.headers(self.master_key)
+        response = self.fulfill(HTTPMethods.GET, url, headers=headers, timeout=self.get_timeout)
+        self._error_handling(response)
+
+        return response.json()
+
+    @requires_key(KeenKeys.MASTER)
+    def get_access_key(self, access_key_id):
+        """
+        Returns details on a particular access key. A master key must be set first.
+
+        :param access_key_id: the 'key' value of the access key to retreive data from
+        """
+        url = "{0}/{1}/projects/{2}/keys/{3}".format(self.base_url, self.api_version, self.project_id,
+                                                     access_key_id)
+        headers = utilities.headers(self.master_key)
+        response = self.fulfill(HTTPMethods.GET, url, headers=headers, timeout=self.get_timeout)
+        self._error_handling(response)
+
+        return response.json()
+
+    def _build_access_key_dict(self, access_key):
```

Comment:
It seems this could be static, i.e. no need for the `self` param

Reply:
I thought it was following a pre-existing pattern, and maybe it is but... we should probably run a cleanup pass through these functions that don't need `self` at some point in the near future -- including this one.

Reply:
I expected the Codacity Code Analysis check to complain about it. Maybe that only runs on master? Not sure.

Reply:
It did complain, but the same pattern exists elsewhere.  I'll plan on having this merged in about 18 hours, but before I do so, I'll probably remove this erroneous `self` and leave other cleanup for later.